platform/core/src/services/MeasurementService/MeasurementService.js

```diff
@@ -256,6 +282,93 @@ class MeasurementService {
     }
   }
 
+  /**
+   * Add a raw measurement into a source so that it may be
+   * Converted to/from annotation in the same way. E.g. import serialized data
+   * Of the same form as the measurement source.
+   * @param {MeasurementSource} source The measurement source instance.
+   * @param {string} definition The source definition you want to add the measuremnet to.
+   * @param {object} data The data you wish to add to the source.
+   * @param {function} toMeasurementSchema A function to get the `data` into the same shape as the source definition.
+   */
+  addRawMeasurement(source, definition, data, toMeasurementSchema) {
```

Comment:
My kingdom for unit tests.
This method makes a lot of sense 👍

Reply:
I made a note in a jira ticket that this shares a lot of functionality with `addOrUpdate` and should be refactored (OHIF-216).

Yeah your unit tests needs a bunch of work most likely 😅 .